src/lib.rs

```diff
@@ -342,12 +355,23 @@ impl<R: gimli::Reader> Context<R> {
 
     /// Find the source file and line corresponding to the given virtual memory address.
     pub fn find_location(&self, probe: u64) -> Result<Option<Location<'_>>, Error> {
-        for unit in self.find_units(probe) {
-            if let Some(location) = unit.find_location(probe, &self.sections)? {
-                return Ok(Some(location));
-            }
+        let sections = &self.sections;
+        let mut unit_iter = self.find_units_range(probe, probe + 1);
+
+        match unit_iter.next() {
+            Some(unit) => unit.find_location(probe, sections),
+            None => Ok(None),
```

Comment:
Can you change this function back to exactly how it was on `master`? It's important that we loop through the units here if required. Unfortunately we don't seem to have any tests that check this case...

Reply:
Ah, right.